src/components/DayPicker/DayPicker.tsx

```diff
@@ -0,0 +1,278 @@
+  formatDate = (day) =>
+    `${day.getMonth() + 1}/${day.getDate()}/${day.getFullYear()}`,
+  parseDate,
+  formatMonthTitle = LocaleUtils.formatMonthTitle,
+  formatWeekdayShort = LocaleUtils.formatWeekdayShort,
+  renderDay,
+  ...textFieldProps
+}) => {
+  if (
+    !textFieldProps.label &&
+    !textFieldProps['aria-label'] &&
+    process.env.NODE_ENV === 'development'
+  ) {
+    throw new Error(
+      'If a "label" is not provided to DayPicker, please provide "aria-label".'
+    );
+  }
+
+  // @ts-ignore This is throwing a bogus type error.
+  const classes = useStyles({});
+  const theme = useTheme();
+
+  const [isCalendarOpen, setIsCalendarOpen] = React.useState(false);
+
+  const calendarRef = React.useRef<ReactDayPicker>(null);
+
+  const wrapperRef = React.useRef<HTMLDivElement>(null);
+  useClickOutside(wrapperRef, () => setIsCalendarOpen(false), []);
+
+  /**
+   * Holds the intermediary text input, which may not yet be a valid date
+   * string.
+   *
+   * `undefined` means the value has either:
+   *   - never been set, or
+   *   - was just set via the date picker
+   *
+   * an empty string means that:
+   *   - the user has set the value to an empty string
+   */
+  const [intermediateInput, setIntermediateInput] = React.useState<
+    string | undefined
+  >(undefined);
+
+  const _onTextChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (!parseDate) {
+      return;
+    }
+
+    onTextChange?.(e.target.value);
+
+    const date = parseDate(e.target.value);
+
+    if (!date) {
+      setIntermediateInput(e.target.value);
+      return;
+    }
+
+    setIntermediateInput(undefined);
+    onDayChange?.(date);
+    calendarRef.current?.showMonth(date);
+  };
+
+  const onDayClick = (day: Date) => {
+    setIntermediateInput(undefined);
+    onDayChange?.(day);
+    onTextChange?.(formatDate(day));
+  };
+
+  return (
+    <div ref={wrapperRef}>
+      <TextField
+        startAdornment={<Calendar />}
+        {...textFieldProps}
+        value={intermediateInput ?? (value ? formatDate(value) : undefined)}
+        className={clsx(textFieldProps.className, {
+          [classes.textFieldNonEditable]: !parseDate,
+        })}
+        onChange={_onTextChange}
+        onFocus={(e) => {
+          setIsCalendarOpen(true);
+          textFieldProps.onFocus?.(e);
+        }}
+      />
+      {isCalendarOpen && (
+        <ReactDayPicker
+          ref={calendarRef}
+          className={classes.dayPicker}
+          modifiers={{
+            selected: value,
+          }}
+          modifiersStyles={{
+            selected: {
+              backgroundColor:
+                selectedBackgroundColor || theme.palette.primary.main,
+            },
+          }}
+          navbarElement={(props) => (
+            <div className={classes.navBar}>
+              <ChevronLeft
+                cursor={'pointer'}
+                onClick={() => props.onPreviousClick()}
+              />
```

Comment:
Since these are raw SVGs with `onClick`, I'm wondering how you'd feel about making these wrapped in button tags:

```suggestion
              <button aria-label="Previous month" onClick={() => props.onPreviousClick()}>
                <ChevronLeft aria-hidden />
              </button>
```

I don't remember if you'd need to override any default styling with `<button`, but essentially it'd be a plain, unstyled click target with an `aria-label` so that screenreaders can access them (and get read the `aria-label` to know what they do). May be able to use the `<ButtonUnstyled` component here!

```suggestion
              <ButtonUnstyled aria-label="Previous month" onClick={() => props.onPreviousClick()}>
                <ChevronLeft aria-hidden />
              </ButtonUnstyled >
```

Reply:
I'm an a11y noob -- thanks for this tip!

I'll add this, but also add customization for the aria-labels (someone might want to localize them).